src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -572,6 +578,8 @@ public void Dispose()
             }
         }
 
+        public bool InitialReadCompleted => _initialRead != null && _initialRead.IsCompleted;
```

Comment:
Based on current usage, it shouldn't be.  I could change it to an assert if you'd prefer.